### PR TITLE
ocotpus and arpack-ng: fix arpack bug and add libs to arpack-ng

### PR DESCRIPTION
**Pull request by davydden**
also rename the variant to be consistent with other packages

The bug was introduced in https://github.com/LLNL/spack/pull/2819

@danielsjensen1 ping.

### CODE REVIEW

**Comment by davydden** [comment]
flake8 errors appears to be false positives:
```
var/spack/repos/builtin/packages/arpack-ng/package.py:130: [F811] redefinition of unused 'install' from line 96
```

File: var/spack/repos/builtin/packages/arpack-ng/package.py [hunk 1]
```diff
@@ -80,6 +80,19 @@ class ArpackNg(Package):
...
     depends_on('mpi', when='+mpi')
 
+    @property
+    def libs(self):
+        # TODO: do we need spec['arpack-ng:parallel'].libs ?
+        # query_parameters = self.spec.last_query.extra_parameters
+        libraries = ['libarpack']
+
+        if '+mpi' in self.spec:
+            libraries = ['libparpack'] + libraries
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/arpack-ng/package.py]: I would just use the append method.

Comment (davydden) [inline var/spack/repos/builtin/packages/arpack-ng/package.py]: but that's prepending, `parpack` should come first before `arpack`.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/arpack-ng/package.py]: Ah, I see. Thanks

**Comment by adamjstewart** [comment]
I would add F811 to the flake8 ignore file. As for the failing unit test, that concerns me. Must be a transient bug.  Perhaps the fixture is failing? Anyway, it's not related to this PR and it should pass if you rerun the test, although we should fix it someday. @alalazo since you're more familiar with pytest can you take a look at this one?

**Comment by alalazo** [comment]
> I would add F811 to the flake8 ignore file.

I would not. Maybe we can skip it on packages, but for me it was helpful at least a couple of times to find typos in python files (e.g. I found two tests with the same name in one test file some times ago).

> As for the failing unit test, that concerns me. Must be a transient bug. Perhaps the fixture is failing? Anyway, it's not related to this PR and it should pass if you rerun the test, although we should fix it someday. @alalazo since you're more familiar with pytest can you take a look at this one?

Agreed, I'll take a look at the tests in `mirror.py` even though I don't know how much I can get out of it...

**Comment by tgamblin** [comment]
I was also hesitant to add 811 to the global
Exceptions.  But we could modify spack flake8 to add a per-line noqa for it, when @when is used on a function in any package.py.  Thoughts?

**Comment by adamjstewart** [comment]
Agreed. Let's just do it for `package.py` files.

**Comment by adamjstewart** [comment]
Oops

**Comment by davydden** [comment]
wait a second, it looks like there is a bug in the PR...

File: var/spack/repos/builtin/packages/octopus/package.py [hunk 2]
```diff
@@ -93,20 +91,19 @@ def install(self, spec, prefix):
                 '--with-netcdf-include=%s' %
                 spec['netcdf-fortran'].prefix.include,
             ])
-        if '+arpack-ng' in spec:
+        if '+arpack' in spec:
+            arpack_libs = spec['arpack-ng'].libs.joined()
             args.extend([
-                '--with-arpack={0}'.format(arpack.joined()),
-            ])
-        if '+scalapack' in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/octopus/package.py]: this is a bug, fixing....

File: var/spack/repos/builtin/packages/octopus/package.py [hunk 3]
```diff
@@ -56,15 +56,13 @@ class Octopus(Package):
...
     depends_on('netcdf-fortran', when='+netcdf')
-    depends_on('arpack-ng', when='+arpack-ng')
+    depends_on('arpack-ng', when='+arpack')
 
     # optional dependencies:
     # TODO: parmetis, etsf-io, sparskit,
     # feast, libfm, pfft, isf, pnfft
 
     def install(self, spec, prefix):
-        arpack = find_libraries('libarpack', root=spec[
```
Comment (davydden) [inline var/spack/repos/builtin/packages/octopus/package.py]: btw, this is the original issue -- unconditionally query for arpack spec.

**Comment by davydden** [comment]
it should be good now:
```
$ otool -L /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/octopus-6.0-6rorbkc5jo6jzzze2pllxakk3thadab4/bin/octopus
/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/octopus-6.0-6rorbkc5jo6jzzze2pllxakk3thadab4/bin/octopus:
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/netlib-scalapack-2.0.2-kaxsskjja7r2efumwznklcmyfq2npxub/lib/libscalapack.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/arpack-ng-3.4.0-tjsotpbiaw6r763g4r6qi3epsehkhifb/lib/libparpack.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/arpack-ng-3.4.0-tjsotpbiaw6r763g4r6qi3epsehkhifb/lib/libarpack.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/parmetis-4.0.3-newahkm2utweqotqaq4ondhs5x2cgcqg/lib/libparmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/metis-5.1.0-qcvico4siizb4xlbrkkboulhfqhbz7gw/lib/libmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/fftw-3.3.6-pl2-fwc4clc2v45nwctjgnurdjqz4kauawwh/lib/libfftw3_threads.3.dylib (compatibility version 9.0.0, current version 9.6.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/fftw-3.3.6-pl2-fwc4clc2v45nwctjgnurdjqz4kauawwh/lib/libfftw3_mpi.3.dylib (compatibility version 9.0.0, current version 9.6.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/fftw-3.3.6-pl2-fwc4clc2v45nwctjgnurdjqz4kauawwh/lib/libfftw3.3.dylib (compatibility version 9.0.0, current version 9.6.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/openblas-0.2.19-c2gdrjmns3xqq7wmlau6ttkhblfhwje3/lib/libopenblas_sandybridgep-r0.2.19.dylib (compatibility version 0.0.0, current version 0.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/gsl-2.3-fcqncx72hwfoqijz225brveb52i4mzh2/lib/libgsl.19.dylib (compatibility version 23.0.0, current version 23.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/gsl-2.3-fcqncx72hwfoqijz225brveb52i4mzh2/lib/libgslcblas.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/openmpi-2.1.0-qbgot3u3qru2cdncr72nmiqwtyojjzkg/lib/libmpi_usempif08.20.dylib (compatibility version 31.0.0, current version 31.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/openmpi-2.1.0-qbgot3u3qru2cdncr72nmiqwtyojjzkg/lib/libmpi_mpifh.20.dylib (compatibility version 31.0.0, current version 31.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/openmpi-2.1.0-qbgot3u3qru2cdncr72nmiqwtyojjzkg/lib/libmpi.20.dylib (compatibility version 31.0.0, current version 31.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/gcc-6.3.0-kn4b6xtmi7tqxwzz7gzctbtn2efngfhs/lib64/libgfortran.3.dylib (compatibility version 4.0.0, current version 4.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.51.1)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/gcc-6.3.0-kn4b6xtmi7tqxwzz7gzctbtn2efngfhs/lib64/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/gcc-6.3.0-kn4b6xtmi7tqxwzz7gzctbtn2efngfhs/lib64/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
```

**Comment by davydden** [comment]
the only failing part on Travis is now
```
[F811] redefinition of unused 'install' from line 96
```

**Comment by adamjstewart** [comment]
For now you can add a `# noqa` to that line. Someone will have to find a clever way to add it to https://github.com/LLNL/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L52

**Pull request merged**